feat(agents): add some more details on what we want for c++ code (#10091)
nextcloud/desktop: AGENTS.md

- Include a short summary of what changed. *Example:* `fix: prevent crash on empty todo title`.
- **Pull Request**: When the agent creates a PR, it should include a description summarizing the changes and why they were made. If a GitHub issue exists, reference it (e.g., “Closes #123”).

> ## c++ Specifics

Reviewer: Capital "C", right? 😁 

```suggestion
## C++ Specifics
```

Author: done

> The following details are important and only relevant when working on the desktop client parts written in c++ language.

Reviewer: ```suggestion
The following details are important and only relevant when working on the desktop client parts written in C++ language.
```

Author: done

> You should never use std::iostream for input or output handling and rather use Qt QLoggingCategory and related classes.

Reviewer: ```suggestion
You should never use `std::iostream` for input or output handling and rather use Qt's `QLoggingCategory` and related classes.
```

> All logging categories starts with nextcloud identifier.

Reviewer: ```suggestion
All custom logging category names start with our "nextcloud." prefix.
```

Author: done

> Our c++ code is using a pattern known as almost always auto when you should almost always not use explicit data types and rather use the auto keyword.

Reviewer: ```suggestion
Our C++ code is using a pattern known as "almost always". You should almost never use explicit data types but the `auto` keyword for type declarations.
```

Author: done

> Our c++ code should can make use of c++ 20 standard features.

Reviewer: ```suggestion
Our C++ code must make use of C++ 20 standard features whenever possible.
```

Author: done

> We do not use c++ modules and rather use standard include of headers.

Reviewer: ```suggestion
Do not use C++ modules. Use standard header inclusion instead.
```

Author: done
